streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -115,6 +182,36 @@ class ZStream[-R, +E, -M, +B, +A](
       _ <- pull.toManaged_
     } yield ()
 
+  /**
+   * Runs the sink on the stream to produce either the sink's internal state or an error.
+   */
+  def runQuery[R1 <: R, E1 >: E, A1 >: A, B](sink: ZSink[R1, E1, B, A1, Nothing]): ZIO[R1, E1, B] =
```

Comment:
It'd be nice to get variants of all the `run` methods that preserve the `ZManaged` for composition. In fact, the ZIO variants can be implemented in terms of those.

Reply:
Done